### DISCUSSION TITLE
Broke my world like a dumbass

### DISCUSSION THREAD

**Discussion by Nubidubi23** [Pryaxis/TShock#1920] [General]
Last night at 4 am i was playing expert mode 1.4.3 and when i got 4 shot by some random cavern enemy i decided enough was enough and was gonna change it to normal with TShock with the worldmode command. Lo and behold , it was an older release that worked up to 1.4.2. Long story short : 
![image](https://user-images.githubusercontent.com/48162191/82867030-bad62900-9f32-11ea-9688-bfa3b0b78d59.png)
Please help :(

**Comment by hakusaro** [comment]
You could start the server back up right? You can run that command from the console. The error you posted is a client error — if that really did cause the problem you should be able to change it back.

**Reply by Nubidubi23** [reply]
How do i do that? And in case you thought it doesn't get any more idiotic from my part , i deleted the world and pasted it into some random folder in order to "experiment with it" (corrupt it in 12 other ways while crying myself to sleep) , can i still do the backup? If yes , how?

**Reply by hakusaro** [reply]
1. Move the world file back into the folder you got it from.
2. Start TShock. Start the server on the world you want to change.
3. Type `/worldmode expert`.
4. Type `/save`.
5. Type `/exit`.

If the world still doesn’t work you need to load it into TEdit or a similar program that’s capable of fixing corrupted worlds.

**Reply by Nubidubi23** [reply]
Which version do i start? The one that corrupted my world or the new one compatible with 1.4.0.3?

**Reply by Nubidubi23** [reply]
Also , i tried opening with T-Edit and it showed no blocks no npcs no nothing , just blank world with nothing in it

**Reply by hakusaro** [reply]
It doesn’t matter which version you start the server with. Make sure you back up your world.

**Reply by Nubidubi23** [reply]
No matter what i do its still fucked , and T-Edit doesn't work either... wtf do i do :(
Here's the world in case you can do something
[pls fix.zip](https://github.com/Pryaxis/TShock/files/4680843/pls.fix.zip)

**Reply by Nubidubi23** [reply]
and in case you're wondering , the T-Edit worlds are from like 2 weeks ago from when i hadn't even beat skeletron

**Reply by Patrikkk** [reply]
One of the TEDIT world files were dated the 24th of May for last modification. Not sure if that is the real date, but I was able to save it in both normal and expert mode.
https://puu.sh/FOWfb/040aa3b608.zip

![](https://puu.sh/FOWet/b8ce2d17c5.png)
Anyhow, it is always recommended to have your worlds automatically back up, which is a supported feature of TShock. 
The original files are unfortunately 24 kilobytes, which means you won't be able to get any data out of it.

**Reply by Nubidubi23** [reply]
(sorry for the world name) 
so i'm screwed..?

**Reply by Nubidubi23** [reply]
OMG I LOVE YOU IT WORKED YOU SAVED ME OMGOMGOMG

**Reply by Patrikkk** [reply]
Make sure to enable backups this time, 
![](https://puu.sh/FOWYx/a5985d586b.png)
This will create a backup every 30 minutes, and keep them for an hour. (This should only leave 2 files in your backup folder inside /tshock/ folder.
Glad I could help :)

**Comment by Nubidubi23** [comment]
one more thing before i close this thread : 
![image](https://user-images.githubusercontent.com/48162191/82886773-abb1a400-9f4f-11ea-97d4-15e39493b1c9.png)
lol
